apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/FirestoreDbTest.cs

```diff
@@ -133,6 +133,84 @@ public void GetDocumentReferenceFromResourcePath_Invalid(string path)
             Assert.Throws<ArgumentException>(() => db.GetDocumentReferenceFromResourceName(path));
         }
 
+        [Fact]
+        public async Task SnapshotAllAsync()
+        {
+            string docName1 = "projects/proj/databases/db/documents/col1/doc1";
+            string docName2 = "projects/proj/databases/db/documents/col2/doc2";
+            var mock = new Mock<FirestoreClient> { CallBase = true };
+            var request = new BatchGetDocumentsRequest
+            {
+                Database = "projects/proj/databases/db",
+                // Note that we request docName2 twice, as per the caller's request:
+                // the library doesn't perform any elision.
+                Documents = { docName1, docName2, docName2 }
+            };
+            var responses = new[]
+            {
+                // Deliberately not in the requested order
+                new BatchGetDocumentsResponse
+                {
+                    Found = new Document
+                    {
+                        Name = docName2, CreateTime = CreateProtoTimestamp(0, 1), UpdateTime = CreateProtoTimestamp(0, 2),
+                        Fields = { { "Name", CreateValue("Test") } }
+                    },
+                    ReadTime = CreateProtoTimestamp(1, 3)
+                },
+                new BatchGetDocumentsResponse { Missing = docName1, ReadTime = CreateProtoTimestamp(1, 2) },
+            };
+            mock.Setup(c => c.BatchGetDocuments(request, It.IsAny<CallSettings>())).Returns(new FakeDocumentStream(responses));
+            var db = FirestoreDb.Create("proj", "db", mock.Object);
+            var docRef1 = db.Document("col1/doc1");
+            var docRef2 = db.Document("col2/doc2");
+            var results = await db.SnapshotAllAsync(new[] { docRef1, docRef2, docRef2 });
+
+            Assert.Equal(3, results.Count);
+            // Note that this is the first result, not the first - we get them back in request order, not response order.
```

Comment:
The first part of this comment seems like it is missing something.

Reply:
Whoops - will fix, thanks.